### DISCUSSION TITLE
How to test with relative import

### DISCUSSION THREAD

**Discussion by itsmnthn** [vitest-dev/vitest#1285] [Q&A] [answered]
I have `nuxt3_rc_3` project and using `vitest` to test the utilities I wrote for project

`utils/index.ts` imports few constants from `~~/config/constants`

While writing test in `test/utils/index.test.ts` I imported one function to write test when I run test I get this error
```bash
FAIL  test/utils/index.test.ts [ test/utils/index.test.ts ]
Error: [vite-node] Failed to load ~~/config/constants
```

I guess the problem is vitest not reading `./.nuxt/tsconfig.json` for relative path lookup or may be something else wrong please help.

**Comment by itsmnthn** [comment]
Is there a way to define paths?

**Comment by itsmnthn** [comment, marked as answer]
Using `alias` in vite config similar to `tsconfig.json` solves the relative import issue in tests.


```Typescript
// alias.ts
import { resolve } from 'path'

const r = (p: string) => resolve(__dirname, p)

export const alias: Record<string, string> = {
  '~~': r('.'),
  '~~/': r('./'),
  '@@': r('.'),
  '@@/': r('./'),
  assets: r('./assets'),
  public: r('./public'),
  'public/': r('./public/'),
}
```

```Typescript
// vitest.config.ts
import { defineConfig } from 'vite'
import { alias } from './alias'

export default defineConfig({
  root: '.',
  esbuild: {
    tsconfigRaw: '{}',
  },
  resolve: {
    alias,
  },
})
```

**Comment by omonk** [comment]
You can also use [vite-tsconfig-paths](https://vitest.dev/guide/common-errors#cannot-find-module-relative-path) to achieve the same thing 

```typescript
import { defineConfig } from "vitest/config";
import tsconfigPaths from "vite-tsconfig-paths";

export default defineConfig({
  plugins: [tsconfigPaths()],
});
```
